### ISSUE TITLE
GitLab auth shipped behind feature flag, Mattermost to new Auth tables

### ISSUE THREAD

**Issue by jordanh** [ParabolInc/parabol#5750]
## Issue - Enhancement

Let's get https://github.com/ParabolInc/parabol/pull/5594 merged! Here is the work remaining.

Note, before this merge can proceed, https://github.com/ParabolInc/parabol/pull/5550 will need to be merged first

#### Responses to comments

   - Replaced `/oauth/token` path in setupOauth2Provider in packages/server/integrations/IntegrationServerManager.ts:53 with a parameter 
   - `oauthRefreshToken` and `oauthScopes` columns moved to `attributes` JSON column

#### Tests
After adding gitlab feature flag to your user...

   - Running `yarn dev` or `yarn postdeploy` inserts properly GLOBAL scoped GitLab provider to `IntegrationProvider`
   - Can authenticate against GitLab Cloud (gitlab.com)
      - Proper token material added to `IntegrationToken`
   - Can remove GitLab Cloud integration configuration
      -  Token material removed from `IntegrationToken` 
   - Can authenticate against Self-Hosted GitLab (https://gitlab-jordanh.parabol.co/)
      - Proper token material added to `IntegrationToken` 
   - Can remove GitLab Self-hosted Integration configuration
      -  Token material removed from `IntegrationToken` 

#### Cleanup & Merger
   - Spooky 👻  TS errors (why is `Json[]` long longer a known type?) resolved
   - Passed Maintainer review & merged

**Estimated effort:** 21 points ([see CONTRIBUTING.md](https://github.com/ParabolInc/parabol/blob/master/CONTRIBUTING.md#points-and-sizes))

### See also:

- #5564
- #5567

**Comment by mattkrick** [comment]
this merged with the gitlab work.
the exception is gitlab self hosted, we'll handle that in another issue since it'll be much less than 21 points
